File: SDKs/visualcrossing.com/weather/4.5/python/src/sdk/weather_forecast.py
import requests
from sdk.models import operations
from . import utils

class WeatherForecast:
    _client: requests.Session
    _security_client: requests.Session
    _server_url: str
    _language: str
    _sdk_version: str
    _gen_version: str

    def __init__(self, client: requests.Session, security_client: requests.Session, server_url: str, language: str, sdk_version: str, gen_version: str) -> None:
        self._client = client
        self._security_client = security_client
        self._server_url = server_url
        self._language = language
        self._sdk_version = sdk_version
        self._gen_version = gen_version

    
    def get_visual_crossing_web_services_rest_services_weatherdata_forecast(self, request: operations.GetVisualCrossingWebServicesRestServicesWeatherdataForecastRequest) -> operations.GetVisualCrossingWebServicesRestServicesWeatherdataForecastResponse:
        r"""Weather Forecast API
        Provides access to weather forecast information. The forecast is available for up to 15 days at the hourly, 12 hour and daily summary level.
        """
        
        base_url = operations.GET_VISUAL_CROSSING_WEB_SERVICES_REST_SERVICES_WEATHERDATA_FORECAST_SERVERS[0]
        if request.server_url is not None:
            base_url = request.server_url
        
        
        url = base_url.removesuffix("/") + "/VisualCrossingWebServices/rest/services/weatherdata/forecast"
        
        query_params = utils.get_query_params(request.query_params)
        
        client = self._client
        
        r = client.request("GET", url, params=query_params)
        content_type = r.headers.get("Content-Type")

        res = operations.GetVisualCrossingWebServicesRestServicesWeatherdataForecastResponse(status_code=r.status_code, content_type=content_type)
        
        if r.status_code == 200:
            pass

        return res

    

File: SDKs/visualcrossing.com/weather/4.5/typescript/src/sdk/weatherforecast.ts
import { AxiosInstance, AxiosRequestConfig, AxiosResponse, ParamsSerializerOptions } from "axios";
import * as operations from "./models/operations";
import * as utils from "../internal/utils";

export class WeatherForecast {
  _defaultClient: AxiosInstance;
  _securityClient: AxiosInstance;
  _serverURL: string;
  _language: string;
  _sdkVersion: string;
  _genVersion: string;

  constructor(defaultClient: AxiosInstance, securityClient: AxiosInstance, serverURL: string, language: string, sdkVersion: string, genVersion: string) {
    this._defaultClient = defaultClient;
    this._securityClient = securityClient;
    this._serverURL = serverURL;
    this._language = language;
    this._sdkVersion = sdkVersion;
    this._genVersion = genVersion;
  }
  
  /**
   * getVisualCrossingWebServicesRestServicesWeatherdataForecast - Weather Forecast API
   *
   * Provides access to weather forecast information. The forecast is available for up to 15 days at the hourly, 12 hour and daily summary level.
  **/
  getVisualCrossingWebServicesRestServicesWeatherdataForecast(
    req: operations.GetVisualCrossingWebServicesRestServicesWeatherdataForecastRequest,
    config?: AxiosRequestConfig
  ): Promise<operations.GetVisualCrossingWebServicesRestServicesWeatherdataForecastResponse> {
    if (!(req instanceof utils.SpeakeasyBase)) {
      req = new operations.GetVisualCrossingWebServicesRestServicesWeatherdataForecastRequest(req);
    }
    
    let baseURL: string = operations.GetVisualCrossingWebServicesRestServicesWeatherdataForecastServerList[0];
    if (req.serverURL) {
      baseURL = req.serverURL;
    }
    const url: string = baseURL.replace(/\/$/, "") + "/VisualCrossingWebServices/rest/services/weatherdata/forecast";
    
    const client: AxiosInstance = this._defaultClient!;
    const qpSerializer: ParamsSerializerOptions = utils.getQueryParamSerializer(req.queryParams);

    const requestConfig: AxiosRequestConfig = {
      ...config,
      params: req.queryParams,
      paramsSerializer: qpSerializer,
    };
    
    
    const r = client.request({
      url: url,
      method: "get",
      ...requestConfig,
    });
    
    return r.then((httpRes: AxiosResponse) => {
        const contentType: string = httpRes?.headers?.["content-type"] ?? "";

        if (httpRes?.status == null) throw new Error(`status code not found in response: ${httpRes}`);
        const res: operations.GetVisualCrossingWebServicesRestServicesWeatherdataForecastResponse = {statusCode: httpRes.status, contentType: contentType};
        switch (true) {
          case httpRes?.status == 200:
            break;
        }

        return res;
      })
  }

}
